components/director/internal/open_resource_discovery/ord_document.go

```diff
@@ -201,27 +216,36 @@ func (docs Documents) Validate(calculatedBaseURL string, apisFromDB map[string]*
 			}
 		}
 
-		decreaseIndexForDeleting := 0
-		for eventIndex := range eventsIndicesWithEmptyName {
-			deleteIndex := eventIndex - decreaseIndexForDeleting
-			doc.EventResources = append(doc.EventResources[:deleteIndex], doc.EventResources[deleteIndex+1:]...)
-			decreaseIndexForDeleting++
-		}
-
-		for _, vendor := range doc.Vendors {
+		for i, vendor := range doc.Vendors {
 			if err := validateVendorInput(vendor); err != nil {
 				errs = multierror.Append(errs, errors.Wrapf(err, "error validating vendor with ord id %q", vendor.OrdID))
+				invalidVendorIndices = append(invalidVendorIndices, i)
+				continue
 			}
 			if _, ok := vendorIDs[vendor.OrdID]; ok {
 				errs = multierror.Append(errs, errors.Errorf("found duplicate vendor with ord id %q", vendor.OrdID))
 			}
 			vendorIDs[vendor.OrdID] = true
 		}
-		for _, tombstone := range doc.Tombstones {
+
+		for i, tombstone := range doc.Tombstones {
 			if err := validateTombstoneInput(tombstone); err != nil {
 				errs = multierror.Append(errs, errors.Wrapf(err, "error validating tombstone with ord id %q", tombstone.OrdID))
+				invalidTombstoneIndices = append(invalidTombstoneIndices, i)
+				continue
```

Comment:
do we really need this `continue` here?

Reply:
Actually we don't need it, yes, because it will go to the next iteration even without it.